python/dgl/graphbolt/graph_storage/csc_sampling_graph.py

```diff
@@ -300,6 +302,83 @@ def sample_neighbors(
             nodes, fanouts.tolist(), replace, return_eids, probs_or_mask
         )
 
+    def sample_neighbors_for_pairs(
+        self,
+        node_pairs,
+    ):
+        """
+        Sample the neighboring edges of the nodes formed by all connected
+        nodes in the provided node pairs. Return the induced subgraph and the
+        compacted node pairs.
+
+        Parameters
+        ----------
+        node_pairs : Tuple[torch.Tensor] or Dict(etype, Tuple[torch.Tensor])
+            Node pairs representing source-destination edges.
+
+        Returns
+        -------
+        Tuple[node_pairs, CSCSamplingGraph]
+            The compacted node pairs and the created CSCSamplingGraph object.
+        """
+
+        nodes_dict = defaultdict(list)
+        unique_nodes_all_types = []
+        if self.metadata:
+            node_type_to_id = self.metadata.node_type_to_id
+        node_type_offset = self.node_type_offset
+        is_mapping = isinstance(node_pairs, Mapping)
+        # Treat it as a homogeneous graph.
+        if is_mapping:
+            assert (
+                node_type_to_id is not None and node_type_offset is not None
+            ), "Please note that sampling with different edge types is \
+                not supported in a homogeneous graph."
```

Comment:
Is this not going to be supported, or is it just "not yet implemented"?

Reply:
It is not supported.